barn-api-service/modules/barn/routers/barn_router.py

```diff
@@ -11,10 +12,18 @@
 async def barn_sections():
     return await get_barn_sections()
 
+@barn_router.post("/barn_sections/", response_model=BarnSchema)
+async def post_barn_sections(barn_section: BarnInputSchema):
+    return await create_barn_section(barn_section)
+
 @barn_router.get("/horses", response_model=list[HorseSchema])
 async def horses():
     return await get_horses()
 
+@barn_router.post("/horses", response_model=HorseSchema)
+async def post_horses(horses: HorseInputSchema):
```

Comment:
Here you have a variable that's plural `horses` but you're only ever taking in data about a single horse. I think this should be called `horse` instead.